pkg/clusterd/disk.go

```diff
@@ -66,6 +67,24 @@ func DiscoverDevices(executor exec.Executor) ([]*sys.LocalDisk, error) {
 			// go on without udev info
 			logger.Warningf("failed to get udev info for device %s: %+v", d, err)
 		}
+
+		// Test if device has child, if so we skip it and only consider the partitions
+		// which will come in later iterations of the loop
+		// We only test if the type is 'disk', this is a property reported by lsblk
+		// and means it's a parent block device
+		if disk.Type == sys.DiskType {
+			deviceChild, err := sys.ListDevicesChild(executor, d)
```

Comment:
This could also be good place to use `LocalDisk.HasChildren` property.  I don't think its being used in current code base anywhere.  Maybe we can add something like below the `PopulateDeviceInfo` method and get the value for `HasChildren` . 

```
if _, ok := udevInfo["ID_PART_TABLE_TYPE"]; ok {
		if udevInfo["DEVTYPE"] == "disk" {
			disk.HasChildren = true
		}
	}
```
Maybe this way we can skip `ListDevicesChild` method completely.

Reply:
This won't work because you can have a filesystem on a raw device. This doesn't mean it's a child.

I believe your example means: "I'm a device type disk (so I'm not a partition) and I have a filesystem".

Reply:
I might be wrong, but I think it means " I'm a disk device type (so not a partition) but I have partitions table (so I have partitions/children)."  

```
sh-4.4# lsblk   
NAME    MAJ:MIN RM  SIZE RO TYPE MOUNTPOINT
xvda    202:0    0  120G  0 disk 
|-xvda1 202:1    0    1M  0 part 
|-xvda2 202:2    0    1G  0 part /boot
`-xvda3 202:3    0  119G  0 part /sysroot
xvdf    202:80   0  100G  0 disk 


sh-4.4# udevadm info --query=property /dev/xvda   
DEVNAME=/dev/xvda
DEVPATH=/devices/vbd-51712/block/xvda
DEVTYPE=disk
ID_PART_TABLE_TYPE=gpt
ID_PART_TABLE_UUID=88c86f80-2658-4071-b89b-87825a0f94e4
MAJOR=202
MINOR=0
SUBSYSTEM=block
TAGS=:systemd:
USEC_INITIALIZED=10430411


sh-4.4# udevadm info --query=property /dev/xvdf   
DEVNAME=/dev/xvdf
DEVPATH=/devices/vbd-51792/block/xvdf
DEVTYPE=disk
MAJOR=202
MINOR=80
SUBSYSTEM=block
TAGS=:systemd:
USEC_INITIALIZED=1700020720
sh-4.4# 
```

Reply:
However, they are cases where the `udevadm` call could fail and we just continue so this check is not 100% reliable where `lsblk` is.

Reply:
Got it. Thanks for the details.